src/Package/Impl/Publishing/Handlers/RmdPublishHandler.cs

```diff
@@ -25,12 +30,48 @@ internal sealed class RmdPublishHandler : IMarkdownFlavorPublishHandler {
             return true;
         }
 
-        public string GetCommandLine(string inputFile, string outputFilePath, PublishFormat publishFormat, Encoding encoding) {
-            string format = GetDocumentTypeString(publishFormat);
-            string outputFile = Path.GetFileName(outputFilePath);
-            string outputFolder = Path.GetDirectoryName(outputFilePath).Replace('\\', '/');
-            // Run rmarkdown::render
-            return Invariant($"rmarkdown::render('{inputFile}', output_format='{format}', output_file='{outputFile}',  output_dir='{outputFolder}', encoding='cp{encoding.CodePage}')");
+        public async Task PublishAsync(IRSession session, ICoreShell coreShell,  IFileSystem fs, string inputFilePath, string outputFilePath, PublishFormat publishFormat, Encoding encoding) {
+            long rmdBlobId = 0;
+            long outputRmdBlobId = 0;
+
+            try {
+                // Send the markdown to R-Host for processing as a blob
+                rmdBlobId = await session.SendFileAsBlobAsync(fs, inputFilePath);
+                if (rmdBlobId <= 0) {
+                    return;
+                }
+
+                // Perform markdown rendering on the blob
+                JObject jsonObj = await RMarkdownRenderAsync(session, rmdBlobId, outputFilePath, publishFormat, encoding);
+
+                // Markdown results should be in a single blob (file) output. The response for markdown rendering should contain 
+                // the blob id and the output file name.
+                JArray jsonBlobids = (JArray)jsonObj["blob_ids"];
```

Comment:
This is still too much details. Why does publishing have to know all the gory details about JSON, specific property names and so on. 

There is nothing more permanent than temporary changes. Think about deployment of files on F5 or implementation of rtvs::fetch that we need to provide so users can get their files. I'd rather do proper file transfer service.